### PR TITLE
Updated readme and added templates

### PR DESCRIPTION
**Pull request by srallen**
Added some info about the BEM usage in the CSS and a section on writing in ES6 in the readme.

Added templates for issues and pull requests. Definitely open to suggestions here.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -149,23 +149,45 @@ There are some ~~nice~~ **unfortunate** (in hindsight) components to help with a
...
+
+- Native ES6 classes are preferred since eventually `React.createClass()` will be deprecated, however, if the existing component is relying on mixins, then consider using `React.createClass()`.
+
+- Mixins are being deprecated and not supported with native classes, so do not use them in new components.
+
+- Use backticks to import ES6 components into coffeescript components: 
+
+```
+`import NewComponet from './new-component'`
+```
```
Comment (eatyourgreens) [inline README.md]: typo: `NewComponent`

File: README.md [hunk 2]
```diff
@@ -149,23 +149,45 @@ There are some ~~nice~~ **unfortunate** (in hindsight) components to help with a
...
+
+An ESLint configration file is setup in the root of the repository for you to use with your text editor to lint both ES6 and use Airbnb's React style guide.
```
Comment (eatyourgreens) [inline README.md]: typo: `configuration`

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+# Review Checklist
+
+- [ ] Does it work in all major browsers: Firefox, Chrome, Edge, Safari?
+- [ ] Does it work on mobile?
+- [ ] Can you `rm -rf node_modules/ && npm install` and app works as expected?
+- [ ] Did you deploy a staging branch?
+
+## Optional
+
+- [ ] If it's a new component, is it in ES6? Is it clear of warnings from ESLint?
```
Comment (eatyourgreens) [inline .github/PULL_REQUEST_TEMPLATE.md]: Add this?
`[ ] Have you replaced any ChangeListener or PromiseRenderer components with code that updates component state?`

**Comment by eatyourgreens** [comment]
LGTM. Couple of typos and maybe we should mention that we're getting rid of `ChangeListener` and `PromiseRenderer`.

Otherwise, I'm happy to merge this and let people submit suggested changes as PRs.

**Comment by srallen** [comment]
I've fixed the typos and added the checkbox for removing `PromiseRenderer` and `ChangeListener`.

**Pull request merged**